src/nvim/buffer_updates.c

```diff
@@ -237,7 +237,7 @@ void buf_updates_send_changes(buf_T *buf,
   for (size_t i = 0; i < kv_size(buf->update_callbacks); i++) {
     BufUpdateCallbacks cb = kv_A(buf->update_callbacks, i);
     bool keep = true;
-    if (cb.on_lines != LUA_NOREF) {
+    if (cb.on_lines != LUA_NOREF && (cb.preview || !(State & CMDPREVIEW))) {
```

Comment:
Not sure about this version, but that avoided a whole bunch of changes, by passing arguments and so on